### PR TITLE
Added support for dynamic beams & spannable LHS

### PR DESCRIPTION
**Pull request by kushalc**

### CODE REVIEW

File: discodop/pcfg.pyx [hunk 1]
```diff
@@ -374,6 +367,10 @@ cdef _parse_clean(sent, CFGChart_fused chart, Grammar grammar,
         return chart, msg
 
     for span in range(1, lensent + 1):
+        spannable = grammar.emission._spannable(span, prepared=prepared_doc) if grammar.emission \
```
Comment (port8080) [inline discodop/pcfg.pyx]: Where should I look for the definition of
`grammar.emission._spannable`?

File: discodop/pcfg.pyx [hunk 2]
```diff
@@ -321,22 +321,18 @@ cdef class SparseCFGChart(CFGChart):
...
     :param tags: Optionally, a sequence of POS tags to use instead of
             attempting to apply all possible POS tags.
     :param start: integer corresponding to the start symbol that complete
             derivations should be headed by; e.g., ``grammar.toid['ROOT']``.
             If not given, the default specified by ``grammar`` is used.
-    :param beam_beta: keep track of the best score in each cell and only allow
-            items which are within a multiple of ``beam_beta`` of the best score.
-            Should be a negative log probability. Pass ``0.0`` to disable.
-    :param beam_delta: the maximum span length to which beam search is applied.
+
```
Comment (port8080) [inline discodop/pcfg.pyx]: Nit: Add comment about beamer

File: discodop/pcfg.pyx (hunk 1 above)
Comment (kushalc) [inline discodop/pcfg.pyx]: https://github.com/talent-works/backend/pull/832/files#diff-7faf36750e589719f7d874ccc8b40c61R664

**Comment by kushalc** [comment]
All blocking commits addressed; merging ahead of final ack given T-3 days to press launch and need to start DB-wide backfill. Let me know if you have any other requests and I'll try to make them post-merge.

**Pull request merged**